context/src/env/parser.rs

```diff
@@ -64,6 +67,49 @@ pub enum CIPlatform {
     Drone,
 }
 
+#[cfg(feature = "ruby")]
+impl CIPlatform {
+    pub fn to_string(&self) -> &str {
+        match self {
+            CIPlatform::GitHubActions => "GITHUB_ACTIONS",
+            CIPlatform::JenkinsPipeline => "JENKINS_PIPELINE",
+            CIPlatform::CircleCI => "CIRCLECI",
+            CIPlatform::Buildkite => "BUILDKITE",
+            CIPlatform::Semaphore => "SEMAPHORE",
+            CIPlatform::TravisCI => "TRAVIS_CI",
+            CIPlatform::Webappio => "WEBAPPIO",
+            CIPlatform::AWSCodeBuild => "AWS_CODEBUILD",
+            CIPlatform::BitbucketPipelines => "BITBUCKET",
+            CIPlatform::AzurePipelines => "AZURE_PIPELINES",
+            CIPlatform::GitLabCI => "GITLAB_CI",
+            CIPlatform::Drone => "DRONE",
+        }
+    }
+}
```

Comment:
nit: We should use `ci_platform_env_key` here.

Optional: We could use this outside of Ruby too and there's a nice couple of traits that we can implement for it as well: `Into<&str>` and `ToString`

For example:
```suggestion
impl Into<&str> for CIPlatform {
    fn into(self) -> &'static str {
        match self {
            CIPlatform::GitHubActions => ci_platform_env_key::GITHUB_ACTIONS,
            CIPlatform::JenkinsPipeline => ci_platform_env_key::JENKINS_PIPELINE,
            CIPlatform::CircleCI => ci_platform_env_key::CIRCLECI,
            CIPlatform::Buildkite => ci_platform_env_key::BUILDKITE,
            CIPlatform::Semaphore => ci_platform_env_key::SEMAPHORE,
            CIPlatform::TravisCI => ci_platform_env_key::TRAVIS_CI,
            CIPlatform::Webappio => ci_platform_env_key::WEBAPPIO,
            CIPlatform::AWSCodeBuild => ci_platform_env_key::AWS_CODEBUILD,
            CIPlatform::BitbucketPipelines => ci_platform_env_key::BITBUCKET,
            CIPlatform::AzurePipelines => ci_platform_env_key::AZURE_PIPELINES,
            CIPlatform::GitLabCI => ci_platform_env_key::GITLAB_CI,
            CIPlatform::Drone => ci_platform_env_key::DRONE,
        }
    }
}

impl ToString for CIPlatform {
    fn to_string(&self) -> String {
        String::from(Into::<&str>::into(*self))
    }
}

#[cfg(feature = "ruby")]
impl CIPlatform {
    pub fn to_string(&self) -> &str {
        (*self).into()
    }
}
```

Reply:
wasn't sure if we'd want this to be general available. Will update.

Reply:
Just checked and these aren't quire the same. `ci_platform_env_key::JENKINS_PIPELINE` is `BUILD_ID` and not `JENKINS_PIPELINE`. I'm good making this change since it's just an example so far, but it won't necessarily match expectations upfront.